.github/workflows/native.yml

```diff
@@ -0,0 +1,38 @@
+# Goal here is to run all CI that do not require the ESP-IDF environment
+name: CI for Native
+
+on:
+  workflow_call:
+
+env:
+  CARGO_INCREMENTAL: 0
+  RUSTFLAGS: "-D warnings"
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+    - name : Checkout main branch code
+      if: github.event_name != 'pull_request_target'
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
+    - name: Check out PR branch code
+      if: github.event_name == 'pull_request_target'
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 2
+    - uses: dtolnay/rust-toolchain@1.83
+      with:
+        components: clippy
+    - name: Cache Dependencies
+      uses: Swatinem/rust-cache@v2
+    - name: install libudev
+      run: sudo apt update && sudo apt install libudev-dev -y
+    - name: Clippy Micro-RDK workspace
+      run: cargo clippy --workspace --no-deps --locked --all-targets --features native,ota
+    - name: Tests
+      run: cargo test --workspace --tests --no-fail-fast --features native,ota
```

Comment:
I sort of wish every `cargo ...` command we had in the workflows was actually a `make some-target` instead. I'd like there to be one source of truth for what I can build locally that will validate what is going to be built in CI. Without the makefile targets and the CI actions being 1:1, that's hard to do.